### PR TITLE
Check for undefined property in processArray

### LINKED ISSUE #33: "Deleting" whole "database" crashes
I have always used the following code to "delete" a "database", e.g. set it back to `{}`:

```
const jsondb = require('node-json-db');
const db = new jsondb('./mydb.json');

db.push("/id", "1");
db.delete("/");
```

Since 0.7.4, this fails with the following trace:

```
/home/ckrooss/sample/node_modules/node-json-db/lib/utils.js:19
        var match = arrayIndexRegex.exec(property.trim());
                                                  ^
TypeError: Cannot read property 'trim' of undefined
    at Object.processArray (/home/ckrooss/sample/node_modules/node-json-db/lib/utils.js:19:51)
```

Which I could easily fix by adding something like `if (!property) return null;` to processArray.
Would this be an acceptable solution worth a PR or is my use case just plain wrong? 
I can always use `db.push("/", {});`, of course.

### PR DESCRIPTION
**Pull request by ckrooss**
Fix crash upon accessing `undefined.trim` when an empty database is passed to processArray

Fixes #33

### CODE REVIEW

**Comment by Belphemur** [comment]
Hello,

Thanks for the PR and the bug report.

File: lib/utils.js [hunk 1]
```diff
@@ -10,7 +10,8 @@
      * @returns ArrayInfo|null
      */
     const processArray = function (property) {
-
+        if (typeof property === 'undefined') return null;
```
Comment (Belphemur) [inline lib/utils.js]: Can you change this line to follow the JSLint ?

It should be 
```javascript
if (typeof property === 'undefined') {
    return null;
}
```

**Comment by ckrooss** [comment]
Hope this helps. Seems like my jshint didn't pick up the config from the Gruntfile.

**Comment by Belphemur** [comment]
Everything look good to me. Thanks also for adding a unit test for this specific bug :)

I'm merging and doing a release for this.

**Pull request merged**